From 2e009edb4f99a575cd6be5567166ca0787db40c5 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 9 Mar 2014 12:32:19 +0000
Subject: [PATCH 04/12] MythMusic: fix playback from CD only playing the first
 track

We already get the filename passed in the ctor so don't try to get it from
the input. Fixes #12088.
---
 mythplugins/mythmusic/mythmusic/cddecoder.cpp |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/cddecoder.cpp b/mythplugins/mythmusic/mythmusic/cddecoder.cpp
index 4524dcf..fb7e186 100644
--- a/mythplugins/mythmusic/mythmusic/cddecoder.cpp
+++ b/mythplugins/mythmusic/mythmusic/cddecoder.cpp
@@ -177,12 +177,7 @@ bool CdDecoder::initialize()
     if (output())
         output()->PauseUntilBuffered();
 
-    QFile* file = dynamic_cast< QFile* >(input()); // From QIODevice*
-    if (file)
-    {
-        setFilename(file->fileName());
-        m_tracknum = getFilename().section('.', 0, 0).toUInt();
-    }
+    m_tracknum = getFilename().section('.', 0, 0).toUInt();
 
     QMutexLocker lock(&getCdioMutex());
 
-- 
1.7.10.2

